apps/taxes/src/modules/ui/app-breadcrumbs.tsx

```diff
@@ -0,0 +1,61 @@
+import { Breadcrumbs } from "@saleor/apps-ui";
+import { useRouter } from "next/router";
+
+type Breadcrumb = {
+  label: string;
+  href?: string;
+};
+
+const newProviderBreadcrumbs = [
+  {
+    href: "/configuration",
+    label: "Configuration",
+  },
+  {
+    label: "Add provider",
+    href: "/providers",
+  },
+] as Breadcrumb[];
+
+const breadcrumbsForRoute = {
+  "/configuration": [
+    {
+      href: "/configuration",
+      label: "Configuration",
+    },
+  ],
+  "/providers": [...newProviderBreadcrumbs],
+  "/providers/taxjar": [
+    ...newProviderBreadcrumbs,
+    {
+      label: "TaxJar",
+      href: "/providers/taxjar",
+    },
+  ],
+  "/providers/avatax": [
+    ...newProviderBreadcrumbs,
+    {
+      label: "Avatax",
+      href: "/providers/avatax",
+    },
+  ],
+} as Record<string, Breadcrumb[]>;
+
+const useBreadcrumbs = () => {
+  const router = useRouter();
+  const breadcrumbs = breadcrumbsForRoute[router.pathname];
+
+  return breadcrumbs;
+};
+
+export const AppBreadcrumbs = () => {
+  const breadcrumbs = useBreadcrumbs();
+
+  return (
+    <Breadcrumbs>
+      {breadcrumbs?.map((breadcrumb) => (
```

Comment:
If breadcrumbs dont exist, it means app is broken (your made paths wrong). If app is broken, it should not fallback, but throw. Thanks to that you can quickly figure out problem, especially that routes are *not* strictly typed. If you fallback, you will render blank / empty component and bug will be harder to find